status/condition_set.go

```diff
@@ -176,52 +177,47 @@ func (r ConditionSet) SetUnknown(conditionType string) {
 
 // recomputeRootCondition marks the root condition to true if all other dependents are also true.
 func (r ConditionSet) recomputeRootCondition(conditionType string) {
-	if condition := r.findUnhappyDependent(); condition != nil {
-		// Propagate unroot dependent to root condition.
+	if conditions := r.findUnhappyDependents(); len(conditions) > 0 {
+		// Propagate dependents to root condition.
 		r.Set(Condition{
-			Type:    r.root,
-			Status:  condition.Status,
-			Reason:  condition.Reason,
-			Message: condition.Message,
+			Type: r.root,
+			// Set false if any are false
+			Status: lo.Ternary(
+				lo.ContainsBy(conditions, func(condition Condition) bool { return condition.IsFalse() }),
+				metav1.ConditionFalse,
+				metav1.ConditionUnknown,
+			),
+			Reason: "UnhealthyDependent",
+			Message: strings.Join(lo.Map(conditions, func(condition Condition, _ int) string {
+				return fmt.Sprintf("%s=%s:%s", condition.Type, condition.Status, condition.Reason)
+			}), ","),
```

Comment:
Is there a standard for conjoined messages? semicolon a little clearer to me but just a nit
```suggestion
			}), ";"),
```